transport/src/main/java/io/netty/channel/nio/NioIoHandler.java

```diff
@@ -710,6 +721,18 @@ private void select(IoHandlerContext runner, boolean oldWakenUp) throws IOExcept
         }
     }
 
+    private static long millisBeforeDeadline(long selectDeadLineNanos, long currentTimeNanos) {
+        assert selectDeadLineNanos != Long.MAX_VALUE;
+        long nanosBeforeDeadline = selectDeadLineNanos - currentTimeNanos;
+        // Prevent overflow when adding the rounding bias:
+        // if we don't do this, it would appear as the deadline is already reached!
```

Comment:
@normanmaurer this is the actual fix: i was observing a select without any timeout to still call selectNow instead - which is not right